lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1067,7 +1067,14 @@ export class UserAgentApplication {
                 window.location.href = "/";
                 return;
             } else if (currentUrl !== loginRequestUrl) {
-                window.location.href = `${loginRequestUrl}${hash}`;
+                // If loginRequestUrl contains a hash (e.g. Angular routing), process the hash now then redirect to prevent both hashes in url
+                if (loginRequestUrl.indexOf("#") > -1) {
+                    this.logger.info("loginRequestUrl contains hash, processing response hash immediately then redirecting");
+                    this.processCallBack(hash, stateInfo, null);
```

Comment:
If there are two hashes does it always imply angular routing?

Reply:
No, #1422 describes another scenario with a non-msal hash value.

Reply:
@pkanher617 No, that was the problematic behavior that @tnorling fixed.